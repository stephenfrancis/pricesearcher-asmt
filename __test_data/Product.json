{
  "product_id": "34053722_gb19",
  "price": 299,
  "price_history": {
    "2017-12-03": 349,
    "2018-01-01": 299
  },
  "product_name": "Dyson V7 Animal Cordless Stick Vacuum Cleaner",
  "description": "The Dyson V7 Animal cord-free vacuum is engineered to clean all floor types. For deeper carpet cleaning, the direct-drive cleaner head delivers 75% more brush bar power than the Dyson V6 Cord-free vacuum. Up to 30 minutes of fade-free suction (20 minutes with direct drive cleaner head, or mini motorized tool). Powerful suction generated by the Dyson digital motor. Mini motorised tool for tougher tasks. Direct-drive cleaner head for carpets. Up to 30 minutes of fade-free suction (20 minutes with direct drive cleaner head, or mini motorized tool). Max mode up to 6 minutes of higher suction for tougher tasks. Hygienic bin emptying. docking station stores and charges the machine so it's ready to use . 22. 5 volts. 20 mins running time on full charge. Low battery indicator. Ideal for pet hair. Variable power. Total capacity 0. 54 litre. Includes mini motorised tool, soft dusting brush and tools included. Size H124. 3, W21, D25cm. Weight 2. 3kg. Manufacturer's 2 year parts and labour guarantee.",
  "image_url": "https://img.fruugo.com/product/8/24/33566248_max.jpg",
  "ps_category": "Vacuum > Stick",
  "brand": "Dyson"
}
