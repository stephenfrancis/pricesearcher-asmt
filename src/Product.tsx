
import * as React from 'react';
import Button from '@material-ui/core/Button';


interface Props {
  data: any;
}

interface State {}

// Top level app component
export default class Product extends React.Component<Props, State> {

  constructor(props) {
    super(props);
    this.state = {};
  }


  public render() {
    return (
      <div>
        <p>Id: {this.props.data.product_id}</p>
        <p>Name: {this.props.data.product_name}</p>
        <p>Description: {this.props.data.description}</p>
      </div>
    );
  }

}
