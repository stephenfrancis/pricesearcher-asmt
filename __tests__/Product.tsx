
import * as React from "react";
import * as Renderer from "react-test-renderer";
import * as ReactTestUtils from 'react-dom/test-utils';
import { GridListTile, } from '@material-ui/core';

import Product from "../src/Product";


const data = {
  "product_id": "34053722_gb19",
  "price": 299,
  "price_history": {
    "2017-12-03": 349,
    "2018-01-01": 299
  },
  "product_name": "Dyson V7 Animal Cordless Stick Vacuum Cleaner",
  "description": "The Dyson V7 Animal cord-free vacuum is engineered to clean all floor types. For deeper carpet cleaning, the direct-drive cleaner head delivers 75% more brush bar power than the Dyson V6 Cord-free vacuum. Up to 30 minutes of fade-free suction (20 minutes with direct drive cleaner head, or mini motorized tool). Powerful suction generated by the Dyson digital motor. Mini motorised tool for tougher tasks. Direct-drive cleaner head for carpets. Up to 30 minutes of fade-free suction (20 minutes with direct drive cleaner head, or mini motorized tool). Max mode up to 6 minutes of higher suction for tougher tasks. Hygienic bin emptying. docking station stores and charges the machine so it's ready to use . 22. 5 volts. 20 mins running time on full charge. Low battery indicator. Ideal for pet hair. Variable power. Total capacity 0. 54 litre. Includes mini motorised tool, soft dusting brush and tools included. Size H124. 3, W21, D25cm. Weight 2. 3kg. Manufacturer's 2 year parts and labour guarantee.",
  "image_url": "https://img.fruugo.com/product/8/24/33566248_max.jpg",
  "ps_category": "Vacuum > Stick",
  "brand": "Dyson"
};


test("appearance", () => {
  const component = Renderer.create(
    <Product data={data} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot(); // modal closed

  const tile = component.root.findAllByType(GridListTile);
  expect(tile.length).toBe(1); // only 1 tiie component
  // tree.children[0].props.onClick();
  // ReactTestUtils.Simulate.click(tile[0].instance);

  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot(); // modal open
});
