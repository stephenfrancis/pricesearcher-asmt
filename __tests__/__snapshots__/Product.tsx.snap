// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appearance 1`] = `
<div>
  <li
    className="MuiGridListTile-root-1"
    onClick={[Function]}
    style={
      Object {
        "margin": 5,
      }
    }
  >
    <div
      className="MuiGridListTile-tile-2"
    >
      <img
        alt="Dyson V7 Animal Cordless Stick Vacuum Cleaner"
        src="https://img.fruugo.com/product/8/24/33566248_max.jpg"
        style={
          Object {
            "maxHeight": 300,
            "maxWidth": 300,
          }
        }
      />
      <div
        className="MuiGridListTileBar-root-5 MuiGridListTileBar-titlePositionBottom-6 MuiGridListTileBar-rootSubtitle-8"
      >
        <div
          className="MuiGridListTileBar-titleWrap-9"
        >
          <div
            className="MuiGridListTileBar-title-12"
          >
            Dyson V7 Animal Cordless Stick Vacuum Cleaner
          </div>
          <div
            className="MuiGridListTileBar-subtitle-13"
          >
            <span>
              Â£ 
              299
            </span>
          </div>
        </div>
      </div>
    </div>
  </li>
</div>
`;
